easystepper/easystepper.go

```diff
@@ -6,11 +6,21 @@ import (
 	"time"
 )
 
+// StepMode determines the number of coil configurations for a single step
+type StepMode uint8
+
+// Valid values for StepMode
+const (
+	FourStepMode  = 4 // 12-23-34-41
```

Comment:
These should be of StepMode type! I'd also go with a prefix naming style for Mode/type/kind enums like so:
```go
ModeFour  StepMode = 4 // 12-23-34-41
ModeEight StepMode = 8 // 1-12-2-23-3-34-4-41
```
Having the `Mode` prefix will make them appear together in intellisense and documentation.